src/components/Login.js

```diff
@@ -68,9 +69,7 @@ const Login = (props) => {
     props.loginUser(loginSubmit);
 
     // If error length is 0 bush the dashbaord
-    props.error.length === 0
-      ? props.history.push("/dashboard")
-      : console.log("false", props.error.length);
+    props.history.push("/dashboard")
```

Comment:
this is why i moved the push to the thunk.  this line does not wait for line 69 to finish which is an axios call, so it will run no matter what.